server/core/src/main/java/org/infinispan/server/core/transport/NettyTransportConnectionStats.java

```diff
@@ -0,0 +1,139 @@
+   private final String threadNamePrefix;
+   private final AtomicLong totalBytesWritten = new AtomicLong();
+   private final AtomicLong totalBytesRead = new AtomicLong();
+
+   public NettyTransportConnectionStats(EmbeddedCacheManager cacheManager, ChannelGroup acceptedChannels, String threadNamePrefix) {
+      this.cacheManager = cacheManager;
+      this.isGlobalStatsEnabled = cacheManager != null && cacheManager.getCacheManagerConfiguration().globalJmxStatistics().enabled();
+      this.acceptedChannels = acceptedChannels;
+      this.threadNamePrefix = threadNamePrefix;
+   }
+
+   private void increment(AtomicLong base, long bytes) {
+      if (isGlobalStatsEnabled)
+         base.addAndGet(bytes);
+   }
+
+
+   public void incrementTotalBytesWritten(long bytes) {
+      increment(totalBytesWritten, bytes);
+   }
+
+   public void incrementTotalBytesRead(long bytes) {
+      increment(totalBytesRead, bytes);
+   }
+
+   public long getTotalBytesWritten() {
+      return totalBytesWritten.get();
+   }
+
+   public long getTotalBytesRead() {
+      return totalBytesRead.get();
+   }
+
+   private boolean needDistributedCalculation() {
+      if (cacheManager != null) {
+         org.infinispan.remoting.transport.Transport transport = cacheManager.getTransport();
+         return transport != null && transport.getMembers().size() > 1;
+      }
+      return false;
+   }
+
+   private int calculateGlobalConnections() {
+      Cache<Object, Object> cache = cacheManager.getCache();
+      DistributedExecutorService exec = new DefaultExecutorService(cache);
+      try {
+         // Submit calculation task
+         List<CompletableFuture<Integer>> results = exec.submitEverywhere(
+               new ConnectionAdderTask(threadNamePrefix));
+         // Take all results and add them up with a bit of functional programming magic :)
+         return results.stream().mapToInt(f -> {
+            try {
+               return f.get(30, TimeUnit.SECONDS);
+            } catch (InterruptedException | ExecutionException | TimeoutException e) {
+               throw new CacheException(e);
+            }
+         }).sum();
+      } finally {
+         exec.shutdown();
+      }
+   }
+
+   static class ConnectionAdderTask implements Serializable, DistributedCallable<Object, Object, Integer> {
+      private final String serverName;
+
+      Cache<Object, Object> cache;
+
+      ConnectionAdderTask(String serverName) {
+         this.serverName = serverName;
+      }
+
+      @Override
+      public void setEnvironment(Cache<Object, Object> cache, Set<Object> inputKeys) {
+         this.cache = cache;
+      }
+
+      @Override
+      public Integer call() throws Exception {
+         GlobalConfiguration globalCfg = cache.getCacheManager().getCacheManagerConfiguration();
+         String jmxDomain = globalCfg.globalJmxStatistics().domain();
+         MBeanServer mbeanServer = JmxUtil.lookupMBeanServer(globalCfg);
```

Comment:
We should be doing this directly without jumping through a local JMX call...

Reply:
Could you please show me how?

Reply:
Please create a follow up JIRA for this